ImGui 1.92.8 breaks sokol_imgui.h by obseleting ImDrawCallback_ResetRenderState

Sokol commit: `2ef84875`
Dear ImGui: 1.92.8

Compiling `sokol.cpp` with g++ 16.1.1

``` c++
// among other things
#define SOKOL_IMPL

#define IMGUI_DISABLE_OBSOLETE_FUNCTIONS
#include "imgui.h"

#include "util/sokol_imgui.h"
```

I get

```
In file included from src/sokol.cpp:12:
build/_deps/sokol-src/util/sokol_imgui.h: In function ‘void simgui_render()’:
build/_deps/sokol-src/util/sokol_imgui.h:2816:43: error: ‘ImDrawCallback_ResetRenderState’ was not declared in this scope
```

Release [notes of ImGui 1.92.8](https://github.com/ocornut/imgui/releases/tag/v1.92.8) clearly calls out obsoleting this function:

> DrawList: obsoleted `ImDrawCallback_ResetRenderState` in favor of using `ImGui::GetPlatformIO().DrawCallback_ResetRenderState`, which is part of our new standard draw callbacks. (https://github.com/ocornut/imgui/issues/9378) Redirecting the earlier value into the later one when set, so both old and new code should work.

Moving to ImGui 1.92.7 works.

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -1,5 +1,14 @@
 ## Updates
 
+### 21-May-2026
+
+- sokol_imgui.h: fix for removal of the deprecated `ImDrawCallback_ResetRenderState`
+  magic value (only an issue when building with `IMGUI_DISABLE_OBSOLETE_FUNCTIONS`).
+
+  Original issue which introduced the magic value check: https://github.com/floooh/sokol/issues/1000
+  New issue with the breakage report: https://github.com/floooh/sokol/issues/1514
+  Fix PR: https://github.com/floooh/sokol/pull/1515
+
 ### 11-May-2026
 
 - A new header `sokol_framebuffer.h` has been added which provides a 'CPU framebuffer'

--- a/util/sokol_imgui.h
+++ b/util/sokol_imgui.h
@@ -2812,8 +2812,10 @@ SOKOL_API_IMPL void simgui_render(void) {
             ImDrawCmd* pcmd = &cl->CmdBuffer.Data[cmd_index];
             if (pcmd->UserCallback != 0) {
                 // User callback, registered via ImDrawList::AddCallback()
-                // (ImDrawCallback_ResetRenderState is a special callback value used by the user to request the renderer to reset render state.)
-                if (pcmd->UserCallback != ImDrawCallback_ResetRenderState) {
+                // NOTE: once the deprecated ImDrawCallback_ResetRenderState define has
+                // been completely removed from Dear ImGui the below magic value can be removed too
+                const intptr_t deprecated_ImDrawCallback_ResetRenderState_magic_value = (-8);
+                if ((intptr_t)pcmd->UserCallback != deprecated_ImDrawCallback_ResetRenderState_magic_value) {
                     pcmd->UserCallback(cl, pcmd);
                     // need to re-apply all state after calling a user callback
                     sg_reset_state_cache();

